### PR TITLE
Fixes #1581: Uses 32767 as wildcard meta like IC2

### LINKED ISSUE #1581: Validate Item Metadata before entering system or notify user of invalid items in the system
can put it into the ae2 system but cannot take it out 
(shift click also cannot, right click also cannot, left click also cannot)
only can taken out by give 1 that item to me and put it into the crafting table in ae2 and take out the rest

### PR DESCRIPTION
**Pull request by Driphter**
Whenever an ingredient in `...\assets\ic2\config\shaped_recipes.ini` and `...\assets\ic2\config\shapless_recipes.ini` ends in a `@*` IC2 give the ingredient a meta of 32767. When IC2 compares `ItemStack`s it uses 32767 as a meta wildcard.

### CODE REVIEW

**Comment by yueh** [comment]
This is a potentially fix for #1552 not #1581 (which is simply broken on their side).

The main issue here is the method is named `isSameItemType` not `isEqualItemType` and some callers require it to be the same type not just an equal. So changing this can potentially cause new bugs.

Also hardcoding the wildcard value is really bad, as this has already changed in the past.

**Comment by Driphter** [comment]
My mistake on the issue number. I'm not sure how I screwed that up.

Other than that, I agree with you on both points. Would it be reasonable to only compare the metadata of items with `hasSubtypes`? [This is currently how NEI does it.](https://github.com/Chicken-Bones/CodeChickenLib/blob/master/src/codechicken/lib/inventory/InventoryUtils.java#L278)

**Comment by yueh** [comment]
No issue with the number, just in case it gets mixed up somehow.

`hasSubtypes` might be an idea, but I cannot exclude any potientially issues with it just out of my head.
One of the actual lines affected by it is [Platform.java#L1835](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/3a9296f453c986c2dfc2de9133172f1a254bff4a/src/main/java/appeng/util/Platform.java#L1835), so it could also something to be handled in `sameOre()`.
Basically there are a numbers of potential fixes for it. There are even ways to handle it in the first try without needing the `checkFuzzy` branch, but these are really deep changes and I just have not enough time latey to wrap my head around it and which issues can appear from it.
When I need to choose between an existing (small) inconvenience and potentially introducing a dupe bug, I still favor the first option.

**Comment by Driphter** [comment]
Perhaps we could be super-specific and have a check for `hasSubtypes`, `damage == 32767`, and `!isItemStackDamageable` when `output.getClass().getCanonicalName() == "ic2.core.item.ItemIC2"`.

Somewhat fragile but still specific enough to likely be safe. If the class name changes it will just stop working like it is right now. I know it introduces a bit of IC2 jank to the code, but at least it can be separated enough to be understandable.

**Comment by yueh** [comment]
If we have to introduce mod specific code to work around their behaviour, I would honestly just say just ignore it.

Most likely not needed for this case. I just have no idea currently about what the best option is.
